src/utils.js

```diff
@@ -39,12 +41,20 @@ export function isWidget( element ) {
  * * sets `contenteditable` attribute to `true`,
  * * adds custom `getFillerOffset` method returning `null`,
  * * adds `ck-widget` CSS class,
- * * adds custom property allowing to recognize widget elements by using {@link ~isWidget}.
+ * * adds custom property allowing to recognize widget elements by using {@link ~isWidget},
+ * * implements `setVirtualSelection` and `removeVirtualSelection` custom properties to handle virtual selection
+ * on widgets.
  *
  * @param {module:engine/view/element~Element} element
  * @param {Object} [options={}]
  * @param {String|Function} [options.label] Element's label provided to {@link ~setLabel} function. It can be passed as
  * a plain string or a function returning a string.
+ * @param {Function} [setVirtualSelection] Function that will apply virtual selection on widget's element. When not
+ * provided, default behaviour will be used: adding CSS class from
+ * {@link module:engine/conversion/buildmodelconverter~VirtualSelectionDescriptor}.
+ * @param {Function} [removeVirtualSelection] Function that will remove virtual selection on widget's element. When not
+ * provided, default behaviour will be used: removing CSS class from
+ * {@link module:engine/conversion/buildmodelconverter~VirtualSelectionDescriptor}.
```

Comment:
I wasn't sure about those new options initially and I need to agree with @pjasiun now. Let's avoid adding such options and make sure that if someone will want to do something differently there will always be an option (which is not part of this ticket of course).